File: xianggole-manager-web/src/main/java/com/xianggole/manager/controller/BrandController.java
package com.xianggole.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.xianggole.pojo.TbBrand;
import com.xianggole.sellergoods.service.BrandService;

import entity.PageResult;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {
	
	@Reference
	private BrandService brandService;
	
	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	} 
	@RequestMapping("/findPage")
	public PageResult findPage(int pageNum,int pageSize){
		return brandService.findPage(pageNum, pageSize);
	} 
	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand brand){
		try {
			 brandService.add(brand);
			 return new Result("添加成功",true);
		}catch (Exception e){
			e.printStackTrace();
			return new Result("添加失败",false);
		}
		
	} 
	@RequestMapping("/findOne")
	public TbBrand findOne(Long id) {
		TbBrand brand = brandService.findOne(id);
		return brand;
	}
	@RequestMapping("/update")
	public Result update (@RequestBody TbBrand brand) {
		try {
			brandService.update(brand);
			return new Result("更改成功",true);
		}catch (Exception e) {
			return new Result("更改失败",false);
		}
		
	}
	@RequestMapping("/delete")
	public Result delete(Long[] ids) {
		try {
			brandService.delete(ids);
			return new Result("删除成功",true);
		}catch (Exception e) {
			return new Result("删除失败",false);
		}
	}
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand,int page,int size) {
		return brandService.findPage(brand, page,size);
	}
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList(){
		return brandService.selectOptionList();
	}

}

File: xianggole-manager-web/src/main/webapp/js/controller/brandController.js
	app.controller('brandController', function($scope,$controller,brandService) {
		$controller('baseController',{$scope:$scope})
		//查询所有品牌列表
		$scope.findAll = function() {
			brandService.findAll().success(function(resule) {
				$scope.list = resule;
			});
		}
		
	
		//分页
// 		$scope.findPage = function(page,size){
// 			$http.get('../brand/findPage.do?pageNum='+page+'&pageSize='+size).success(
// 			function(response){
// 				$scope.list=response.rows;	
// 				$scope.paginationConf.totalItems=response.total;//更新总记录数
// 			}			
// 	);
// 		}
		//添加 or 更改
		$scope.save = function(){
			var object = null;
			if($scope.entity.id==null){
			  object = brandService.add($scope.entity);
			}else{
				object = brandService.update($scope.entity);
			}
			object.success(function(response){
							if(response.success){
										$scope.reloadList();
									}else{
										alert(response.message);
									}
			   });
			
			
		}
		
		//查询一个商品信息
		$scope.findOne = function(id){
			brandService.findOne(id).success(function(result){
				 $scope.entity = result;
			});
		}
		
		
		//删除
		$scope.delet=function(){
			brandService.delet($scope.selectIds).success(function(result){
				 if(result.success){
				 			$scope.reloadList();
				 		}else{
				 			alert(result.message);
				 		}
			});
		}
		//查询
		$scope.searchEntity={};
		$scope.search=function(page,size){
				brandService.search(page,size,$scope.searchEntity).success(function(result){
					$scope.list=result.rows;	
					$scope.paginationConf.totalItems=result.total;
				});
		}
		
	});